internal/node/generate_build_file.js

```diff
@@ -0,0 +1,166 @@
+  // absolute paths, making them non-hermetic.
+  // See https://github.com/bazelbuild/rules_nodejs/issues/347
+  const excludedFiles = ['.md', '.html', '.pyc'];
+  const nodejsSrcFiles = filterFilesForFilegroup(listFiles(NODE_DIR), [], excludedFiles)
+  const yarnSrcFiles = filterFilesForFilegroup(listFiles(YARN_DIR), [], excludedFiles)
+
+  const binaryExt = IS_WINDOWS ? '.cmd' : '';
+  const buildFile = `#Generated by node_repositories.bzl
+package(default_visibility = ["//visibility:public"])
+exports_files([
+  "run_npm.sh.template",
+  "bin/node_args.sh",
+  "${NODE_DIR}/bin/node",
+  "bin/node${binaryExt}",
+  ${!IS_WINDOWS ? '"bin/node.js",' : ''}
+  "bin/npm${binaryExt}",
+  "bin/npm_node_repositories${binaryExt}",
+  "bin/yarn${binaryExt}",
+  "bin/yarn_node_repositories${binaryExt}",
+  ])
+alias(name = "node", actual = "${NODE_ACTUAL}")
+alias(name = "npm", actual = "${NPM_ACTUAL}")
+alias(name = "yarn", actual = "${YARN_ACTUAL}")
+filegroup(
+  name = "node_runfiles",
+  srcs = [
+    ${!IS_WINDOWS ? '"bin/node.js",' : ''}
+    ${nodejsSrcFiles.map(f => `"${NODE_DIR}/${f}",`).join('\n    ')}
+    ${yarnSrcFiles.map(f => `"${YARN_DIR}/${f}",`).join('\n    ')}
+  ],
+)
+`
+  writeFileSync('BUILD.bazel', buildFile);
+}
+
+/**
+ * Returns an array of all the files under a directory as relative
+ * paths to the directory.
+ */
+function listFiles(rootDir, subDir = '') {
+  const dir = path.posix.join(rootDir, subDir);
+  return fs
+    .readdirSync(dir)
+    .reduce((files, file) => {
+      const fullPath = path.posix.join(dir, file);
+      const relPath = path.posix.join(subDir, file);
+      let stat;
+      try {
+        stat = fs.statSync(fullPath);
+      } catch (e) {
+        if (fs.lstatSync(fullPath).isSymbolicLink()) {
+          // filter out broken symbolic links... these cause fs.statSync(fullPath)
+          // to fail with `ENOENT: no such file or directory ...`
+          return files;
+        }
+        throw e;
+      }
+      if (stat.isFile() && (/^BUILD$/i.test(file) || /^BUILD\.bazel$/i.test(file))) {
+        // Delete BUILD and BUILD.bazel files so that so that files do not cross Bazel package
+        // boundaries. npm packages should not generally include BUILD or BUILD.bazel files
+        // but they may as rxjs does temporarily.
+        fs.unlinkSync(fullPath);
+        return files;
+      }
+      return stat.isDirectory() ? files.concat(listFiles(rootDir, relPath)) :
+        files.concat(relPath);
+    }, []);
+}
+
+/**
+ * A filter function for a bazel filegroup.
+ * @param files array of files to filter
+ * @param allowedExts list of white listed extensions; if empty, no filter is done on extensions;
+ *             '' empty string denotes to allow files with no extensions, other extensions
+ *             are listed with '.ext' notation such as '.d.ts'.
+ */
+function filterFilesForFilegroup(files, allowedExts = [], excludedExts = []) {
```

Comment:
Hmmm. A bit of duplication here in the same repo. Maybe move this function to a `utils.js` in the `internal/node` folder and import it in `internal/npm_install/generate_build_file.js`?